subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/cntr.c

```diff
@@ -11,21 +11,73 @@
 
 #include "hal/debug.h"
 
+#if !defined(CONFIG_BT_CTLR_NRF_GRTC)
 #ifndef NRF_RTC
 #if defined(CONFIG_SOC_COMPATIBLE_NRF54LX)
 #define NRF_RTC NRF_RTC10
 #else /* !CONFIG_SOC_COMPATIBLE_NRF54LX */
 #define NRF_RTC NRF_RTC0
 #endif /* !CONFIG_SOC_COMPATIBLE_NRF54LX */
 #endif /* !NRF_RTC */
+#endif /* CONFIG_BT_CTLR_NRF_GRTC */
 
 static uint8_t _refcount;
 
 void cntr_init(void)
 {
+#if defined(CONFIG_BT_CTLR_NRF_GRTC)
+#if defined(CONFIG_BT_CTLR_NRF_GRTC_START)
+	NRF_GRTC->MODE = (GRTC_MODE_SYSCOUNTEREN_Disabled <<
+			  GRTC_MODE_SYSCOUNTEREN_Pos) &
+			 GRTC_MODE_SYSCOUNTEREN_Msk;
+
+	NRF_GRTC->TASKS_CLEAR = 1U;
+
+#if defined(CONFIG_BT_CTLR_NRF_GRTC_KEEPRUNNING)
+	NRF_GRTC->KEEPRUNNING =
+		(GRTC_KEEPRUNNING_REQUEST1_Active <<
+		 GRTC_KEEPRUNNING_REQUEST1_Pos) &
+		GRTC_KEEPRUNNING_REQUEST1_Msk;
+	NRF_GRTC->TIMEOUT = 0U;
+	NRF_GRTC->INTERVAL = 0U;
+	NRF_GRTC->WAKETIME = 4U;
+#endif /* CONFIG_BT_CTLR_NRF_GRTC_KEEPRUNNING */
+
+	NRF_GRTC->CLKCFG = ((GRTC_CLKCFG_CLKSEL_LFXO <<
+			     GRTC_CLKCFG_CLKSEL_Pos) &
+			    GRTC_CLKCFG_CLKSEL_Msk) |
+			   ((GRTC_CLKCFG_CLKFASTDIV_Min <<
+			     GRTC_CLKCFG_CLKFASTDIV_Pos) &
+			    GRTC_CLKCFG_CLKFASTDIV_Msk);
+#endif /* CONFIG_BT_CTLR_NRF_GRTC_START */
+
+	NRF_GRTC->EVENTS_COMPARE[10] = 0U;
+	NRF_GRTC->INTENSET1 = GRTC_INTENSET1_COMPARE10_Msk;
+
+#if defined(CONFIG_BT_CTLR_NRF_GRTC_START)
+	NRF_GRTC->MODE = ((GRTC_MODE_SYSCOUNTEREN_Enabled <<
+			   GRTC_MODE_SYSCOUNTEREN_Pos) &
+			  GRTC_MODE_SYSCOUNTEREN_Msk) |
+#if defined(CONFIG_BT_CTLR_NRF_GRTC_AUTOEN_CPUACTIVE)
+			 ((GRTC_MODE_AUTOEN_CpuActive <<
+			   GRTC_MODE_AUTOEN_Pos) &
+			  GRTC_MODE_AUTOEN_Msk) |
+#endif /* CONFIG_BT_CTLR_NRF_GRTC_AUTOEN_CPUACTIVE */
+#if defined(CONFIG_BT_CTLR_NRF_GRTC_AUTOEN_DEFAULT)
+			 ((GRTC_MODE_AUTOEN_Default <<
+			   GRTC_MODE_AUTOEN_Pos) &
+			  GRTC_MODE_AUTOEN_Msk) |
+#endif /* CONFIG_BT_CTLR_NRF_GRTC_AUTOEN_DEFAULT */
+			 0U;
+
+	NRF_GRTC->TASKS_START = 1U;
```

Comment:
Also a couple of magic numbers here. The naming of the fields also makes it hard to determine whether `1` means a numeric value, or a boolean.

Reply:
some should be replaced now.